### PR TITLE
Test CRDT semantics with TestBucket_Merge

### PR DESCRIPTION
**Pull request by tsenart**
This PR adds a test to verify that the CRDT semantics of a Bucket are implemented correctly.

PTAL @peterbourgon

### CODE REVIEW

**Comment by tsenart** [comment]
Also, @CBaquero, if you have some time :-)

File: bucket_test.go [hunk 1]
```diff
@@ -58,3 +59,50 @@ func TestBucket_Take(t *testing.T) {
...
+	// - Idempotence (a+a=a)
+	//
+	for i := 0; i < 10000; i++ { // Test 10000 random permutations of Merge order.
+		rng.Shuffle(len(buckets), func(i, j int) {
+			buckets[i], buckets[j] = buckets[j], buckets[i]
+		})
+
+		random := buckets[rng.Int()%len(buckets)]
+		for _, bucket := range buckets {
+			random.Merge(&bucket)
```
Comment (peterbourgon) [inline bucket_test.go]: To test idempotence, it might be nice to add e.g.

```go
if (rng.Int() % 100) > 66 {
    random.Merge(&bucket) // again
}
```

Comment (tsenart) [reply bucket_test.go]: Since we start with a random Bucket and merge it with itself already, is this necessary?

Comment (peterbourgon) [reply bucket_test.go]: Ah, so you do exercise idempotence with n=1. I don't know! Perhaps not, but perhaps it's worth exaggerating e.g. network retransmissions, for comedic effect.

Comment (tsenart) [reply bucket_test.go]: Alright, let's be more explicit.

**Review by peterbourgon — APPROVED** [review]
Seems to do what it says on the tin.

**Pull request merged**